float32 x
float32 y
float32 z
int32 index
float64 dist
string entityName